packages/nimble-components/src/wafer-map/modules/prerendering.ts

```diff
@@ -131,13 +132,14 @@ export class Prerendering {
     }
 
     private calculateOpacity(
-        selectedValue: string,
-        highlightedValues: Readonly<string[]>
+        dieTags?: string[],
+        highlightedTags?: string[]
     ): number {
-        return highlightedValues.length > 0
-            && !highlightedValues.some(dieValue => dieValue === selectedValue)
-            ? this.nonHighlightedOpacity
-            : 1;
+        if (!highlightedTags || highlightedTags.length === 0) {
+            return 1;
+        }
+        const tagsMatch = dieTags?.some(dieTag => highlightedTags.some(highlightedTag => dieTag === highlightedTag));
```

Comment:
This does not look very efficient ... I do not know where the `dieTags `or the `highlightedTags ` come from, but did you consider packing them into a set or a map instead of arrays ? It would make this kind of searches much easier and/or faster.

Reply:
I had a similar thought but one discussion point was the ability to for example specify a unique tag for each die to be able to address them: https://github.com/ni/nimble/pull/1678#discussion_r1408282049

It's a contrived example but in that scenario caching dies by tag where the cache is larger than the full set of dies (1 per die and then additional tags) might be costly as well. 

Agree that the assumption that highlightedTags and dieTags is small for now is probably a good starting point and we can consider additional optimizations as the use-case comes up (unless we know of concerns now).